man/systemd-gpt-auto-generator.xml

```diff
@@ -50,19 +50,47 @@
 
-    <para>This generator will only look for the root partition on the same physical disk where the EFI System
-    Partition (ESP) is located. Note that support from the boot loader is required: the EFI variable
-    <varname>LoaderDevicePartUUID</varname> of the <constant>4a67b082-0a4c-41cf-b6c7-440b29bb8c4f</constant>
-    vendor UUID is used to determine from which partition, and hence the disk, from which the system was
-    booted. If the boot loader does not set this variable, this generator will not be able to detect the root
-    partition. See the <ulink url="https://systemd.io/BOOT_LOADER_INTERFACE">Boot Loader Interface</ulink>
-    for details.</para>
-
-    <para>Similarly, this generator will only look for the other partitions on the same physical disk as the
-    root partition. In this case, boot loader support is not required. These partitions will not be searched
-    for on systems where the root file system is distributed on multiple disks, for example via btrfs RAID.
+    <para>When run in the initial RAM disk (initrd) this generator can automatically search for the root file
+    system. Specifically:
+
+    <itemizedlist>
+      <listitem><para>It will look for the root partition on the same physical disk where the EFI System
+      Partition (ESP) is located. Note that support from the boot loader is required for this to work: the
+      EFI variable <varname>LoaderDevicePartUUID</varname> of the
+      <constant>4a67b082-0a4c-41cf-b6c7-440b29bb8c4f</constant> vendor UUID is used to determine from which
+      partition (and hence disk) the system was booted. If the boot loader does not set this variable, this
+      generator will not be able to detect the root partition. See the <ulink
+      url="https://systemd.io/BOOT_LOADER_INTERFACE">Boot Loader Interface</ulink> for
+      details.</para></listitem>
+
+      <listitem><para>Alternatively, it will look for the root file system on a loopback block device whose
+      <literal>.lo_name</literal> field is set to one of the literal strings <literal>rootdisk</literal> or
+      <literal>rootdisk.raw</literal>. This field can be set via <command>losetup</command>'s
+      <option>--loop-ref=</option> string. For images downloaded via
+      <citerefentry><refentrytitle>systemd-import-generator</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+      make sure to set the <literal>blockdev</literal> option and set the local name string to
+      <literal>rootdisk</literal> to achieve this effect. Note that discovery of the root file system on
+      loopback block devices like this is only done if <literal>root=gpt-auto</literal> is specified
+      explicitly on the kernel command line, unlike the discovery based on the boot loader reported ESP which
+      is also enabled if no <literal>root=</literal> parameter is specified at all.</para></listitem>
+    </itemizedlist>
```

Comment:
This conflates udev builtin's gpt-auto symlink creation and what the generator does, which is a bit misleading. Better to state this relies on `/dev/gpt-auto-root`, which might have different sources honored by udev.